socorro/external/boto/crashstorage.py

```diff
@@ -61,116 +153,118 @@ def __init__(self, config, namespace="", quit_check_callback=None):
-            wait_time_generator=wait_time_generator,
-            module_logger=LOGGER,
-        )
+    def save_raw_crash(self, raw_crash, dumps, crash_id):
+        """Save raw crash data to S3 bucket.
 
-    @staticmethod
-    def do_save_raw_crash(conn, raw_crash, dumps, crash_id):
+        A raw crash consists of the raw crash annotations and all the dumps that came in
+        the crash report. We need to save the raw crash file, a dump names file listing
+        the dumps that came in the crash report, and then each of the dumps.
+
+        """
         if dumps is None:
             dumps = MemoryDumpsMapping()
 
-        raw_crash_data = conn._convert_mapping_to_string(raw_crash).encode("utf-8")
-        conn.submit(crash_id, "raw_crash", raw_crash_data)
+        path = build_keys("raw_crash", crash_id)[0]
+        raw_crash_data = dict_to_str(raw_crash).encode("utf-8")
+        self.conn.save_file(path, raw_crash_data)
 
-        dump_names_data = conn._convert_list_to_string(dumps.keys()).encode("utf-8")
-        conn.submit(crash_id, "dump_names", dump_names_data)
+        path = build_keys("dump_names", crash_id)[0]
+        dump_names_data = list_to_str(dumps.keys()).encode("utf-8")
+        self.conn.save_file(path, dump_names_data)
 
         # We don't know what type of dumps mapping we have. We do know,
         # however, that by calling the memory_dump_mapping method, we will get
         # a MemoryDumpMapping which is exactly what we need.
         dumps = dumps.as_memory_dumps_mapping()
-        for dump_name, dump in iteritems(dumps):
+        for dump_name, dump in dumps.items():
             if dump_name in (None, "", "upload_file_minidump"):
                 dump_name = "dump"
-            conn.submit(crash_id, dump_name, dump)
-
-    def save_raw_crash(self, raw_crash, dumps, crash_id):
-        """Save raw_crash, dump_names, and dumps."""
-        retryable_save = self.retryable(self.do_save_raw_crash)
-        with self.connection_source() as conn:
-            retryable_save(conn, raw_crash=raw_crash, dumps=dumps, crash_id=crash_id)
-
-    @staticmethod
-    def do_save_processed(conn, processed_crash):
-        crash_id = processed_crash["uuid"]
-        data = conn._convert_mapping_to_string(processed_crash).encode("utf-8")
-        conn.submit(crash_id, "processed_crash", data)
+            path = build_keys(dump_name, crash_id)[0]
+            self.conn.save_file(path, dump)
 
     def save_processed(self, processed_crash):
-        """Save processed_crash."""
-        retryable_save = self.retryable(self.do_save_processed)
-        with self.connection_source() as conn:
-            retryable_save(conn, processed_crash=processed_crash)
+        """Save the processed crash file."""
+        crash_id = processed_crash["uuid"]
+        data = dict_to_str(processed_crash).encode("utf-8")
+        path = build_keys("processed_crash", crash_id)[0]
+        self.conn.save_file(path, data)
```

Comment:
I find this easier to read than the old `do_save_processed` / `save_processed` code. Nice change!